stability: nil pointer panic in raft

Seen on beta cluster node 1, running 6827d40

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xea6d8d]

goroutine 3372971757 [running]:
panic(0x15549c0, 0xc420010060)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/cockroachdb/cockroach/cli.initBacktrace.func2(0x15549c0, 0xc420010060)
        /go/src/github.com/cockroachdb/cockroach/cli/backtrace.go:98 +0xca
github.com/cockroachdb/cockroach/util/stop.(*Stopper).Recover(0xc420362750)
        /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:173 +0x56
panic(0x15549c0, 0xc420010060)
        /usr/local/go/src/runtime/panic.go:458 +0x243
github.com/coreos/etcd/raft.(*raft).appendEntry(0xc422621860, 0xc4335926e0, 0x1, 0x1)
        /go/src/github.com/coreos/etcd/raft/raft.go:419 +0xdd
github.com/coreos/etcd/raft.(*raft).becomeLeader(0xc422621860)
        /go/src/github.com/coreos/etcd/raft/raft.go:509 +0x21a
github.com/coreos/etcd/raft.stepCandidate(0xc422621860, 0x6, 0x7, 0x6, 0x12094, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/coreos/etcd/raft/raft.go:763 +0xaa3
github.com/coreos/etcd/raft.(*raft).Step(0xc422621860, 0x6, 0x7, 0x6, 0x12094, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/coreos/etcd/raft/raft.go:579 +0x305
github.com/coreos/etcd/raft.(*RawNode).Step(0xc43e4352c0, 0x6, 0x7, 0x6, 0x12094, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/coreos/etcd/raft/rawnode.go:175 +0xbc
github.com/cockroachdb/cockroach/storage.(*Store).handleRaftMessage.func1(0xc43e4352c0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/storage/store.go:2490 +0x61
github.com/cockroachdb/cockroach/storage.(*Replica).withRaftGroupLocked(0xc42339cb40, 0xc46a34d9a0, 0xc42339cbb0, 0xc42339cbb0)
        /go/src/github.com/cockroachdb/cockroach/storage/replica.go:364 +0xa0
github.com/cockroachdb/cockroach/storage.(*Replica).withRaftGroup(0xc42339cb40, 0xc46a34d9a0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/storage/replica.go:373 +0x8a
github.com/cockroachdb/cockroach/storage.(*Store).handleRaftMessage(0xc42024a000, 0xc4205aefc0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/storage/store.go:2491 +0xd96
github.com/cockroachdb/cockroach/storage.(*Store).(github.com/cockroachdb/cockroach/storage.handleRaftMessage)-fm(0xc4205aefc0, 0xc420398690, 0x1)
        /go/src/github.com/cockroachdb/cockroach/storage/store.go:1086 +0x34
github.com/cockroachdb/cockroach/storage.(*RaftTransport).RaftMessage.func1.1.1(0x2010880, 0xc42ae4f3f0, 0xc420357e40, 0xc423010788, 0xc423010790)
        /go/src/github.com/cockroachdb/cockroach/storage/raft_transport.go:145 +0xe4
github.com/cockroachdb/cockroach/storage.(*RaftTransport).RaftMessage.func1.1()
        /go/src/github.com/cockroachdb/cockroach/storage/raft_transport.go:149 +0x45
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc420362750, 0xc428fbcb10)
        /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x7d
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x66
```

--- a/GLOCKFILE
+++ b/GLOCKFILE
@@ -46,7 +46,7 @@ github.com/cockroachdb/crlfmt f4be2332630e8f33c3262c0d5f0d13d733c8bedf
 github.com/cockroachdb/pq 44a6473ebbc26e3af09fe57bbdf761475c2c9f7c
 github.com/cockroachdb/stress 029c9348806514969d1109a6ae36e521af411ca7
 github.com/codahale/hdrhistogram 3a0bb77429bd3a61596f5e8a3172445844342120
-github.com/coreos/etcd 27459425faab1a28ac4d0063eebcea6e91c18339
+github.com/coreos/etcd c9cc1efb67476729242584e00790f9f7f069a1c3
 github.com/cpuguy83/go-md2man a65d4d2de4d5f7c74868dfa9b202a3c8be315aaa
 github.com/docker/distribution fbe6e8d212ed880cf7f7c9f876ba9b15c8221c5f
 github.com/docker/docker 7248742ae7127347a52ab9d215451c213b7b59da

